crates/native-api/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+//! This module provides functionality related to `cairo-native`.
+//!
+//! Currently, it includes:
+//!  - Compiling Sierra contract classes into `cairo-native` executors.
+
+use cairo_lang_starknet_classes::contract_class::ContractClass;
+use cairo_native::executor::AotContractExecutor;
+use starknet_api::contract_class::SierraVersion;
+
+/// Compiles a given Sierra [`ContractClass`] into an [`AotContractExecutor`] for `cairo-native` execution.
+#[must_use]
+pub fn compile_contract_class(contract_class: &ContractClass) -> AotContractExecutor {
+    let sierra_program = contract_class
+        .extract_sierra_program()
+        .expect("extraction should succeed");
+
+    let sierra_version = extract_sierra_version(contract_class);
+
+    AotContractExecutor::new(
+        &sierra_program,
+        &contract_class.entry_points_by_type,
+        sierra_version.clone().into(),
```

Comment:
I don't think we officially support any scarb version that produces incompatible sierra, but as a sanity check, maybe we should validate the version somewhere.

I think native only supports sierra from 1.7.0 upwards.

Maybe minimal scarb version in foundry should be bumped to lowest scarb that supports that version? (Still there's a problem of contracts from network in forking, we should ensure we don't compile incompatible ones).